docs/scheduler-perf/scheduler-perf.md

```diff
@@ -0,0 +1,856 @@
+                                         'host1', request_spec=req,
+                                         filter_properties={})
+        end = time.time()
+        time_call = ((end - start) / float(NUM_TESTS)) * 1000
+        real_time = time_call - self.loop_time
+
+        self.file.write(f'Scheduler create snapshot: {real_time}ms\n')
+```
+
+## Testing with services
+
+Only the volume creation operation was tested in a running system
+
+### Hardware
+
+The tests were done using an old HP ProLiant DL360 Gen9 machine with:
+- CentOS 9 Stream.
+- 2 x Intel Xeon E5-2686 V4 - 18-Core 36-Threads 2.30GHz (3.00GHz Boost,
+  45MB Cache, 145W TDP)
+- 6 x 64GB|PC4-2400-L|x4|Quad Rank
+
+### Environment preparation
+
+OpenStack was deployed in OpenShift using the `install_yaml` repository on a CRC
+system.
+
+The CRC system was deployed with:
+```bash
+$ cd install_yamls/devsetup
+$ PULL_SECRET=~/pull-secret.txt CPUS=26 MEMORY=300000 DISK=150 make crc
+```
+
+Then `iscsid` and the LVM VG were created using the following manifest:
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+    service: cinder
+  name: 99-master-cinder-enable-iscsid
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - enabled: true
+        name: iscsid.service
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+    service: cinder
+    component: cinder-volume
+  name: 99-master-cinder-lvm-losetup
+spec:
+  config:
+    ignition:
+      version:  3.2.0
+    Systemd:
+      Units:
+      - name: cinder-lvm-losetup.service
+        enabled: true
+        Contents:  |
+          [Unit]
+          Description=Cinder LVM losetup
+          DefaultDependencies=no
+          Conflicts=umount.target
+          Requires=lvm2-monitor.service systemd-udev-settle.service
+          Before=local-fs.target umount.target
+          After=var.mount lvm2-monitor.service systemd-udev-settle.service
+
+          [Service]
+          Environment=VG_GB_SIZE=10G
+          Environment=LOOPBACK_FILE=/var/home/core/cinder-volumes
+          Type=oneshot
+          ExecStart=bash -c "if [[ ! -e ${LOOPBACK_FILE} ]]; then /bin/truncate -s ${VG_GB_SIZE} ${LOOPBACK_FILE} && /sbin/vgcreate cinder-volumes `/sbin/losetup --show -f ${LOOPBACK_FILE}`; else /sbin/losetup -f ${LOOPBACK_FILE}; fi"
+          ExecStop=bash -c "/sbin/losetup -d `/sbin/losetup -j ${LOOPBACK_FILE} -l -n -O NAME`"
+          RemainAfterExit=yes
+
+          [Install]
+          WantedBy=local-fs-pre.target
+```
+
+After the node was rebooted the usual steps were followed to deploy things:
+```
+cd install_yamls/devsetup
+eval $(crc oc-env)
+# Link kubectl to oc since we don't have kubectl installed
+ln -s oc `dirname \`which oc\``/kubectl
```

Comment:
I suppose there's no harm including these lines, but they are extraneous. Nothing described in this document uses the kubectl command.

Reply:
Makefile targets use it and won't work if it's not there.
Although this code I added a while back should take care of it... https://github.com/openstack-k8s-operators/cinder-operator/blob/53e4d6a340a78012f2d2a09d1b37d2bbbd17f48a/Makefile#L78

So maybe in my tests I forgot to do the `eval $(crc oc-env)` part before calling `make`...

Reply:
Silly me, I added that to the cinder operator, not the install-yamls repository, so that's why it is necessary.